NCM_APIv2_class/v2_example_class.py

```diff
@@ -0,0 +1,137 @@
+"""
+NOTE: Cradlepoint does not develop, maintain, or support NCM API applications.
+Applications are the sole responsibility of the developer.
+
+WARNING: NCM API Applications can introduce security and other potential issues
+when not carefully engineered. Test your code thoroughly before deploying it to
+production devices. This can affect production devices and data!
+"""
+
+import json
+import os
+import requests
+import logging
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+from datetime import datetime
+
+
+class NCM_APIv2(object):
+
+    headers = {
+        'X-ECM-API-ID': '',
+        'X-ECM-API-KEY': '',
+        'X-CP-API-ID': '',
+        'X-CP-API-KEY': '',
+        'Accept': '*/*',
+        'Content-Type': 'application/json',
+    }
+
+    def get_data(self, url, json_output):
+
+        resp_data = {'data': []}
+
+        while url:
+
+            with requests.Session() as s:
+
+                retries = Retry(total=5,  # Total number of retries to allow.
+                                backoff_factor=2,
+                                status_forcelist=[408,  # 408 Request Timeout
+                                                  500,  # 500 Internal Server Error
+                                                  502,  # 502 Bad Gateway
+                                                  503,  # 503 Service Unavailable
+                                                  504,  # 504 Gateway Timeout
+                                                  ],
+                                )
+                a = requests.adapters.HTTPAdapter(max_retries=retries)
+                s.mount('https://', a)
+
+                try:
+                    logging.info(url)
+                    print(url)
```

Comment:
You don't need print() when you are using logging (and logging is the correct thing to use in a module meant to be used by other programs).